resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -612,11 +612,14 @@ private[yarn] class YarnAllocator(
             val message = "Container killed by YARN for exceeding physical memory limits. " +
               s"$diag Consider boosting ${EXECUTOR_MEMORY_OVERHEAD.key}."
             (true, message)
+          case exit_status if NOT_APP_AND_SYSTEM_FAULT_EXIT_STATUS.contains(exit_status) =>
```

Comment:
I would prefer not to have it as a separate case but just a new if around `handleResourceAllocationFailure` and as NOT_APP_AND_SYSTEM_FAULT_EXIT_STATUS is introduced it would make sense to separate it from the huge match on exitStatus. This way it would be easier to follow when it is really triggered (one should not check all the previous case branches then consider this condition with contains). That way values like     ContainerExitStatus.SUCCESS from the set would be really used.

Reply:
yeah I agree this should be cleaned up we already handle cases above that are in the NOT_APP_AND_SYSTEM_FAULT_EXIT_STATUS set.

Reply:
also after this gets rearranged, I'd leave a comment in here pointing to the code in hadoop you linked to on the jira.

Reply:
Updated. But I'm not sure about:

> That way values like ContainerExitStatus.SUCCESS from the set would be really used.

this part. @attilapiros